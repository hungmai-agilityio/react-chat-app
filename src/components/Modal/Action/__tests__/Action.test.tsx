import { fireEvent, render, screen } from '@testing-library/react';

// FontAwesome
import { faArrowRight } from '@fortawesome/free-solid-svg-icons';

//Components
import ModalAction from '@/components/Modal/Action';

describe('Modal Action', () => {
  const onClickMock = jest.fn();

  const defaultProps = {
    avatar: '',
    onChange: onClickMock,
    name: '',
    icon: faArrowRight
  };

  beforeEach(() => {
    jest.fn().mockClear();
  });

  it('triggers file upload when avatar is clicked', () => {
    render(<ModalAction {...defaultProps} />);
    const avatarElement = screen.getByTestId('avatar');
    const inputFile = screen.getByTestId('file-input') as HTMLInputElement;

    fireEvent.click(avatarElement);
    const file = new File(['dummy content'], 'example.png', {
      type: 'image/png'
    });

    fireEvent.change(inputFile, { target: { files: [file] } });

    expect(inputFile.files?.[0]).toBeTruthy();
    expect(inputFile.files?.[0].name).toBe('example.png');
  });

  it('should render with btnName and button is clicked', () => {
    render(
      <ModalAction {...defaultProps} btnName="Save" name='Test' onClick={onClickMock} />
    );

    const button = screen.getByRole('button', { name: 'Save' });
    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalled();
  });

    it('should render match to snapshot - Default', () => {
      const { container } = render(
        <ModalAction
          {...defaultProps}
        />
      );
      expect(container).toMatchSnapshot();
    });

    it('should render match to snapshot - Button click', () => {
      const { container } = render(
        <ModalAction
          {...defaultProps}
          btnName="Save"
          onClick={onClickMock}
        />
      );
      expect(container).toMatchSnapshot();
    });
});
