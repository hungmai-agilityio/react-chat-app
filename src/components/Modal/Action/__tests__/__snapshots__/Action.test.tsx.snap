// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Action should render match to snapshot - Button click 1`] = `
<div>
  <div
    class="relative mt-5"
  >
    <div
      class="absolute left-1/2 transform -translate-x-1/2"
    >
      <div
        class="text-primary bg-gray-300 flex items-center justify-center font-semibold rounded-lg w-avatar-lg h-avatar-lg text-8xl"
        data-testid="avatar"
      />
      <input
        accept="image/*"
        class="hidden"
        data-testid="file-input"
        id="upload"
        type="file"
      />
    </div>
    <div
      class="absolute top-32 left-1/2 transform -translate-x-1/2"
    >
      <div
        class="mt-4 w-72"
      >
        <div
          class="mb-3"
        >
          <div
            class="relative"
          >
            <input
              class="w-full mt-1.5 outline-none p-3 focus:border-primary h-10 border-b border-gray-400 text-center"
              placeholder="Chat name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="absolute top-lg right-xs"
    >
      <button
        class="font-semibold hover:brightness-90 bg-tertiary cursor-not-allowed border hover:filter-none text-stone-700 w-40 h-9 bg-primary text-white"
        disabled=""
      >
        Save
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right ml-2"
          data-icon="arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal Action should render match to snapshot - Default 1`] = `
<div>
  <div
    class="relative mt-5"
  >
    <div
      class="absolute left-1/2 transform -translate-x-1/2"
    >
      <div
        class="text-primary bg-gray-300 flex items-center justify-center font-semibold rounded-lg w-avatar-lg h-avatar-lg text-8xl"
        data-testid="avatar"
      />
      <input
        accept="image/*"
        class="hidden"
        data-testid="file-input"
        id="upload"
        type="file"
      />
    </div>
    <div
      class="absolute top-32 left-1/2 transform -translate-x-1/2"
    >
      <div
        class="mt-4 w-72"
      >
        <div
          class="mb-3"
        >
          <div
            class="relative"
          >
            <input
              class="w-full mt-1.5 outline-none p-3 focus:border-primary h-10 border-b border-gray-400 text-center"
              placeholder="Chat name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
